### ISSUE TITLE
Inconsistent behaviour while attempting to update the MedicinalProductDefinition with the same content

### ISSUE THREAD

**Issue by rafalbroll-datapharm** [microsoft/fhir-server#3735]
**Describe the bug**
I'm experiencing inconsistent behaviour while I'm attempting to update the MedicinalProductDefinition with unmodified payload (meaning: i'm sending over the content i've got from the GET endpoint, to the PUT endpoint, without any modification to that content). 
It seems like when the MedicinalProductDefinition is at version 1, an attempt of update with unmodified content will not increase the version.
However, when the version is 2 or greater, the FHIR Server will keep increasing the version on every PUT request, even if the payload doesn't contain any difference.

**FHIR Version?**
R4B, [3.4.342]

**Data provider?**
SQL Server

**To Reproduce**
1. Make sure you got some MedicinalProductDefinition in your FHIR Server, which is at version 1 (meta.versionId = 1) 
2. Get the definition using endpoint `GET https://<serverUrl>/MedicinalProductDefinition/<id>`, keep the response
3. Call the endpoint `PUT https://<serverUrl>/MedicinalProductDefinition/<id>` using the response from step 2 as a payload.
4. Enforce the version increment, eg. by running **step 3**. with product's name changed
5. repeat **step 2**
6. repeat **step 3**

**Expected behavior**
While steps 2-3 and 5-6 are doing exactly same job, the FHIR response should be consistent. Probably it should always prevent incrementing the versionId as the payload doesn't contain any changes in compare to the actual data.

**Actual behavior**
The versionId gets increased after step 6

**Comment by brendankowitz** [comment]
Thanks for reporting @rafalbroll-datapharm, we do have this bug scheduled in an upcoming sprint.

Duplicate of AB#112948

**Comment by brendankowitz** [comment]
Hi @rafalbroll-datapharm, this issue is fixed in release https://github.com/microsoft/fhir-server/releases/tag/release%2F3.4.361 and higher.
